### PR TITLE
fix: move documentation to READMEs of packages

### PR DESCRIPTION
**Pull request by daKmoR**
This moves all documentation into the READMEs of each package.
They are just symlinked to the docs folder so vuepress will properly display them.

Before we merge this we should definitely improve some more in the content of these READMEs.

@thepassle @LarsDenBakker @motss @bashmish pls a little help :)

e.g. just add commits to this Branch/PR (once done I will cleanup the history and merge)

### CODE REVIEW

File: docs/recommendations/automate.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
 # Automate
 
-tbd
+Having a ci running test for you is really helpful as it gives valueable insights especially for merge requests.
```
Comment (thepassle) [inline docs/recommendations/automate.md]:
```suggestion
Having a ci running test for you is really helpful as it gives valuable insights especially for merge requests.
```

File: packages/generator-open-wc/README.md [hunk 2]
```diff
@@ -1,9 +1,39 @@
...
 
 We want to provide a good set of default on how to vasilitate your web component.
 
 [![CircleCI](https://circleci.com/gh/open-wc/open-wc.svg?style=shield)](https://circleci.com/gh/open-wc/open-wc)
 [![BrowserStack Status](https://www.browserstack.com/automate/badge.svg?badge_key=M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)](https://www.browserstack.com/automate/public-build/M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
+
+## Usage
+
+This generator is based on [yoeman](http://yeoman.io/). To us it you have 2 ways.
```
Comment (thepassle) [inline packages/generator-open-wc/README.md]:
```suggestion
This generator is based on [yeoman](http://yeoman.io/). To us it you have 2 ways.
```

File: packages/generator-open-wc/README.md [hunk 3]
```diff
@@ -1,9 +1,39 @@
...
+
+```bash
+npm i -g yoeman
```
Comment (thepassle) [inline packages/generator-open-wc/README.md]:
```suggestion
npm i -g yeoman
```

File: packages/generator-open-wc/README.md [hunk 4]
```diff
@@ -1,9 +1,39 @@
...
+  - open-wc:vanilla-bare
+  - open-wc:lint-eslint
+  - open-wc:publish-storybook
+  - open-wc:testing-bare
+  - open-wc:testing-karma
+  - open-wc:testing-karma-bs
+  - open-wc:tools-circleci
+- open-wc:testing-wallaby
+
+For what they actually do pls see the details for each system at [open-wc](https://open-wc.org/)
```
Comment (thepassle) [inline packages/generator-open-wc/README.md]:
```suggestion
For what they actually do please see the details for each system at [open-wc](https://open-wc.org/)
```

File: packages/semantic-dom-diff/README.md [hunk 5]
```diff
@@ -1,9 +1,30 @@
-# Open Web Component Recommendations to compare dom and shadowDom trees
+# Semantic Dom Diff
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/semantic-dom-diff/README.md]:
```suggestion
We want to provide a good set of default on how to facilitate your web component.
```

File: packages/testing-helpers/README.md [hunk 6]
```diff
@@ -1,9 +1,87 @@
-# Open Web Component Recommendations testing helpers
+# Testing Helpers
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/testing-helpers/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/testing-helpers/README.md [hunk 7]
```diff
@@ -1,9 +1,87 @@
...
 
 [![CircleCI](https://circleci.com/gh/open-wc/open-wc.svg?style=shield)](https://circleci.com/gh/open-wc/open-wc)
 [![BrowserStack Status](https://www.browserstack.com/automate/badge.svg?badge_key=M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)](https://www.browserstack.com/automate/public-build/M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
+
+In order to efficient test webcomponent you will need some helpers to register and instanciate them.
```
Comment (thepassle) [inline packages/testing-helpers/README.md]:
```suggestion
In order to efficient test webcomponent you will need some helpers to register and instantiate them.
```

File: packages/testing-helpers/README.md [hunk 8]
```diff
@@ -1,9 +1,87 @@
...
+
+::: tip Info
+This is part of the default [open-wc](https://open-wc.org/) recommendation
+:::
+
+## Test a custom element
+```js
+import { fixture } from '@open-wc/testing-helpers';
+
+it('can instanciate an element', async () => {
```
Comment (thepassle) [inline packages/testing-helpers/README.md]:
```suggestion
it('can instantiate an element', async () => {
```

File: packages/testing-helpers/README.md [hunk 9]
```diff
@@ -1,9 +1,87 @@
...
+  const el = await fixture('<my-el foo="bar"></my-el>');
+  expect(el.getAttribute('foo')).to.equal('bar');
+}
+```
+
+## Test a custom element with properties
+```js
+import { html, litFixture } from '@open-wc/testing-helpers';
+
+it('can instanciate an element with properties', async () => {
```
Comment (thepassle) [inline packages/testing-helpers/README.md]:
```suggestion
it('can instantiate an element with properties', async () => {
```

File: packages/testing-karma-bs/README.md [hunk 10]
```diff
@@ -1,9 +1,58 @@
-# Open Web Component Recommendations for karma using browserstack
+# Testing via Browserstack
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/testing-karma-bs/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/testing-karma/README.md [hunk 11]
```diff
@@ -1,6 +1,6 @@
-# Open Web Component Recommendations for karma
+# Testing with karma
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/testing-karma/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

**Comment by daKmoR** [comment]
so I added at least the minimum documentation for everything now... at least I think

you can take a look at the preview url:
https://deploy-preview-61--nervous-stonebraker-1f429b.netlify.com/

so if someone want's to take a quick lock to make sure we don't say anything stupid would be nice

File: packages/testing-wallaby/README.md [hunk 12]
```diff
@@ -1,9 +1,29 @@
-# Open Web Component Recommendations for wallaby
+# Testing in IDE via wallaby
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/testing-wallaby/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/testing/README.md [hunk 13]
```diff
@@ -1,9 +1,76 @@
-# Open Web Component Recommendations for testing
+# Testing
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/testing/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/webpack/README.md [hunk 14]
```diff
@@ -1,9 +1,25 @@
-# Open Web Component Recommendations for webpack
+# Webpack Helpers
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to facilitate your web component.
```
Comment (thepassle) [inline packages/webpack/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/webpack/README.md [hunk 15]
```diff
@@ -1,9 +1,25 @@
...
 
 [![CircleCI](https://circleci.com/gh/open-wc/open-wc.svg?style=shield)](https://circleci.com/gh/open-wc/open-wc)
 [![BrowserStack Status](https://www.browserstack.com/automate/badge.svg?badge_key=M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)](https://www.browserstack.com/automate/public-build/M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
+
+If you need support to use `import.meta.url` within webpack this is a minimal loader to suppor it.
```
Comment (thepassle) [inline packages/webpack/README.md]:
```suggestion
If you need support to use `import.meta.url` within webpack this is a minimal loader to support it.
```

File: packages/chai-dom-equals/README.md [hunk 16]
```diff
@@ -1,9 +1,55 @@
-# Open Web Component Recommendations Chai Plugin to compare dom and shadowDom trees
+# Testing Chai Dom Equals
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/) Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/chai-dom-equals/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/eslint-config/README.md [hunk 17]
```diff
@@ -1,9 +1,47 @@
-# Open Web Component Recommendations for eslint
+# Linting ESLint
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/) Recommendations [open-wc](https://open-wc.org/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/eslint-config/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/generator-open-wc/README.md [hunk 18]
```diff
@@ -1,9 +1,39 @@
-# Open Web Component Recommendations Generator
+# Generator Open WC
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/) Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/generator-open-wc/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: packages/storybook/README.md [hunk 19]
```diff
@@ -1,9 +1,48 @@
-# Open Web Component Recommendations for storybook
+# Demos via storybook
 
-> Part of Open Web Component Recommendations [open-wc](https://github.com/open-wc/open-wc/)
+> Part of Open Web Component Recommendation [open-wc](https://github.com/open-wc/open-wc/)
 
 We want to provide a good set of default on how to vasilitate your web component.
```
Comment (thepassle) [inline packages/storybook/README.md]:
```suggestion
We want to provide a good set of defaults on how to facilitate your web component.
```

File: docs/guide/README.md [hunk 20]
```diff
@@ -15,11 +15,15 @@ npx -p yo -p generator-open-wc -c 'yo open-wc:vanilla'
...
 - [Automate](/recommendations/automate.html)
 
 ## General Help Section
-- [JavaScript](/help/js.html)
\ No newline at end of file
+- [JavaScript](/help/js.html)
+
+## Example
+The [Set-Game Example](https://github.com/open-wc/example-vanilla-set-game/) uses all the above recommendations
+A published storybook you can find here [https://example-set-game-open-wc.netlify.com/](https://example-set-game-open-wc.netlify.com/).
```
Comment (thepassle) [inline docs/guide/README.md]:
```suggestion
You can find a published storybook here: [https://example-set-game-open-wc.netlify.com/](https://example-set-game-open-wc.netlify.com/).
```

File: docs/recommendations/publishing.md [hunk 21]
```diff
@@ -1,3 +1,23 @@
...
+- it's so easy
+- it provides previews for every MR
+- it supports custom domain
+- ...
+
+## Manual Setup
+
+So just head over to [netlify](https://www.netlify.com/) and register.
+Then select your github account and repository.
+If you follow these recommedations all you need to do is
```
Comment (thepassle) [inline docs/recommendations/publishing.md]:
```suggestion
If you follow these recommendations all you need to do is
```

File: docs/recommendations/publishing.md [hunk 22]
```diff
@@ -1,3 +1,23 @@
...
+- Build command: `npm run site:build`
+- Publish directory: `_site`
+
+And you are all set.
+
+## Example
+The [Set-Game Example](https://github.com/open-wc/example-vanilla-set-game/) has the default publishing via storybook on netlify.
+The finished page you can see at [https://example-set-game-open-wc.netlify.com/](https://example-set-game-open-wc.netlify.com/).
```
Comment (thepassle) [inline docs/recommendations/publishing.md]:
```suggestion
You can see the finished page here: [https://example-set-game-open-wc.netlify.com/](https://example-set-game-open-wc.netlify.com/).
```

File: packages/chai-dom-equals/README.md [hunk 23]
```diff
@@ -1,9 +1,55 @@
...
 
 [![CircleCI](https://circleci.com/gh/open-wc/open-wc.svg?style=shield)](https://circleci.com/gh/open-wc/open-wc)
 [![BrowserStack Status](https://www.browserstack.com/automate/badge.svg?badge_key=M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)](https://www.browserstack.com/automate/public-build/M2UrSFVRang2OWNuZXlWSlhVc3FUVlJtTDkxMnp6eGFDb2pNakl4bGxnbz0tLUE5RjhCU0NUT1ZWa0NuQ3MySFFWWnc9PQ==--86f7fac07cdbd01dd2b26ae84dc6c8ca49e45b50)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
+
+Comparing dom literally is usually not what you want when testing components.
+Additional when using ShadyDOM and/or ShadyCSS there will be additional classes you are not interested in.
```
Comment (thepassle) [inline packages/chai-dom-equals/README.md]:
```suggestion
Additionally, when using ShadyDOM and/or ShadyCSS there will be additional classes you are not interested in.
```

File: packages/storybook/README.md [hunk 24]
```diff
@@ -1,9 +1,48 @@
...
+
+Create stories within the `stories` folder.
+
+```bash
+npm run storybook:start
+```
+
+## Example
+The [Set-Game Example](https://github.com/open-wc/example-vanilla-set-game/) has the default publishing via storybook on netlify.
+The finished page you can see at https://example-set-game-open-wc.netlify.com/.
```
Comment (thepassle) [inline packages/storybook/README.md]:
```suggestion
You can see the finished page at: https://example-set-game-open-wc.netlify.com/.
```

**Pull request merged**